.github/workflows/schedule_tests.yml

```
@@ -37,6 +37,36 @@ jobs:
       - name: Run tests
         run: python tests/runtests.py -v2
 
+  pyc-only:
+    runs-on: ubuntu-latest
+    name: Python with no source files (only pyc files)
+    continue-on-error: true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: 'pip'
+      - name: Install and upgrade packaging tools
+        run: python -m pip install --upgrade pip setuptools wheel
+      - run: python -m pip install .
+      - name: Prepare site-packages
+        run: |
+          SITE_PACKAGE_ROOT=$(python -c 'import site; print(site.getsitepackages()[0])')
+          echo $SITE_PACKAGE_ROOT
+          python -m compileall -b $SITE_PACKAGE_ROOT
+          find $SITE_PACKAGE_ROOT -name '*.py' -print -delete
+      - name: Create project
+        working-directory: ${{ runner.temp }}
+        run: python -m django startproject testproject
+      - name: Check project
+        working-directory: ${{ runner.temp }}/testproject
+        run: |
+          ./manage.py check
+          ./manage.py migrate
```

Can we run our test suite instead? Maybe something like this:
```suggestion
      - run: python -m pip install -r tests/requirements/py3.txt
      - name: Run tests
        run: python tests/runtests.py -v2
```

---

In the context of reviewing PR #17613, I used these instructions to setup a "frozen" env to run the tests. I had to tweak the steps so:

*  `SITE_PACKAGE_ROOT` points to the installed django (so SITE_PACKAGE_ROOT=$(python -c 'import site; print(site.getsitepackages()[0])')"/django")

A full test run failed with:
```
======================================================================
FAIL: test_get_or_create_raises_IntegrityError_plus_traceback (get_or_create.tests.GetOrCreateTestsWithManualPKs.test_get_or_create_raises_IntegrityError_plus_traceback)
get_or_create should raise IntegrityErrors with the full traceback.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/query.py", line 948, in get_or_create
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/query.py", line 649, in get
get_or_create.models.ManualPrimaryKeyTest.DoesNotExist: ManualPrimaryKeyTest matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 354, in execute
sqlite3.IntegrityError: UNIQUE constraint failed: get_or_create_manualprimarykeytest.id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/nessita/fellowship/github-link-docs-392-29942/tests/get_or_create/tests.py", line 238, in test_get_or_create_raises_IntegrityError_plus_traceback
    ManualPrimaryKeyTest.objects.get_or_create(id=1, data="Different")
    ^^^^^^^^^^^^^^^^^
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/query.py", line 955, in get_or_create
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/query.py", line 679, in create
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/base.py", line 860, in save
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/base.py", line 976, in save_base
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/base.py", line 1134, in _save_table
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/base.py", line 1175, in _do_insert
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/query.py", line 1847, in _insert
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1831, in execute_sql
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in execute
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/utils.py", line 100, in _execute
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
  File "/home/nessita/.virtualenvs/django-frozen-no-source/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 354, in execute
django.db.utils.IntegrityError: UNIQUE constraint failed: get_or_create_manualprimarykeytest.id

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/case.py", line 57, in testPartExecutor
    yield
  File "/usr/lib/python3.11/unittest/case.py", line 623, in run
    self._callTestMethod(testMethod)
    ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/case.py", line 579, in _callTestMethod
    if method() is not None:
    ^^^^^^^^^^^^^^^^^
  File "/home/nessita/fellowship/github-link-docs-392-29942/tests/get_or_create/tests.py", line 241, in test_get_or_create_raises_IntegrityError_plus_traceback
    self.assertIn("obj.save", formatted_traceback)
    ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/case.py", line 1140, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
    ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/case.py", line 703, in fail
    raise self.failureException(msg)
    ^^^^^^^^^^^^^^^^^
AssertionError: 'obj.save' not found in <traceback>
----------------------------------------------------------------------
Ran 16275 tests in 284.372s

FAILED (failures=1, skipped=1064, expected failures=5)
```

---

> SITE_PACKAGE_ROOT points to the installed django (so SITE_PACKAGE_ROOT=$(python -c 'import site; print(site.getsitepackages()[0])')"/django")